### PR TITLE
refactor: split files

### PR DESCRIPTION
**Pull request by jly36963**
## Description

- Split functions into separate files
- Group files by kind
- Imports include file extensions 
  - [ESM import docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import)

### CODE REVIEW

**Comment by p9f** [comment]
I would really like to try to keep the tests close to the source ([locality of behaviour mindset](https://htmx.org/essays/locality-of-behaviour/)), do you think we could consider ways to achieve it?

One idea would be to group the utilities by types and then in each folder have all the source + tests:
```
native-utilities/
  join.ts
  join.test.ts
  ...
word-utilities/
  camelCase.ts
  camelCase.test.ts
  ...
object-utilities/
  camelKeys.ts
  camelKeys.test.ts
  deepCamelCase.ts
  ...
```
That might help to make [the new naming convention](https://github.com/gustavoguichard/string-ts/issues/44) clearer.
we could also merge the `.type.test.ts` and `.test.ts` files - as they will be quite small because of the split (see [this](https://github.com/gustavoguichard/string-ts/pull/48#issuecomment-1752007610))

**Comment by gustavoguichard** [comment]
I agree with the comments from @p9f .
Also.. I kind of hate camel cased filenames. I'd rather keep them in kebab case bc it works best in file system world - I've has troubles with that more than once. Do you have a strong reason to use them camel cased?

**Comment by jly36963** [comment]
Couldn't think of.a good name for the `additional` folder

File: tsconfig.json [hunk 1]
```diff
@@ -1,8 +1,5 @@
 {
-  "include": [
-    "src/**/*.ts",
-    "src/**/*.tsx"
-  ],
+  "include": ["src/**/*.ts", "src/**/*.tsx"],
```
Comment (gustavoguichard) [inline tsconfig.json:2]:
```suggestion
  "include": ["src/**/*.ts"],
```

**Comment by gustavoguichard** [comment]
> I couldn't think of.a better name for the `additional` folder

I think the words function belongs to that folder... maybe `utils` (which is also quite broad) sounds better?

**Comment by gustavoguichard** [comment]
> move toLowerCase and toUpperCase to the native folder
Agreed

For the words method, I believe it should belong to the same folder as `truncate` and `reverse` from #71 which are not casing functions nor native ones but popular string utilities from other libs.

**Comment by jly36963** [comment]
For sure, will make those changes during lunch today. If you can think of any other feedback, let me know.
I'm hoping to get this merged quickly so as to avoid a lot of merge conflicts

**Comment by gustavoguichard** [comment]
@jly36963 I'll stop my activity until this PR is merged to avoid conflicts =)

**Review by gustavoguichard — APPROVED** [review]
LGTM

**Pull request merged**